recipes/grpc/all/conanfile.py

```diff
@@ -80,14 +80,18 @@ def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             tools.check_min_cppstd(self, 11)
 
+        if tools.is_apple_os(self.settings.os) and "arm" in self.settings.arch:
+            raise ConanInvalidConfiguration("gRPC is not supported on M1 Mac due to limitations in protobuf support.")
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
-        # Fix the protoc search path for cross compiling
-        tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "protobuf.cmake"),
-                "find_program(_gRPC_PROTOBUF_PROTOC_EXECUTABLE protoc)",
-                "find_program(_gRPC_PROTOBUF_PROTOC_EXECUTABLE protoc PATHS ENV PATH NO_DEFAULT_PATH)"
-        )
-        if tools.Version(self.version) >= "1.39.0":
+        if tools.Version(self.version) < "1.42.0":
```

Comment:
I would prefer moving to build step, but I'm okay for this PR.

Reply:
Thanks a lot! Will do this with a next change...